org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ProjectUtils.java

```diff
@@ -77,16 +77,8 @@ public static boolean isJavaProject(IProject project) {
 		return hasNature(project, JavaCore.NATURE_ID);
 	}
 
-	public static boolean isMavenProject(IProject project) {
```

Comment:
i will be cautious to remove public helpers from the utility since it may be used by other extensions.

Reply:
@testforstephen definitely cautious am I. The whole idea of improving IBuildSupport is to move the logic of working with particular build tool to the specific plugin. If we leave those methods in the utils class then we are leaving one more place for the maven/gradle logic. Which is bad, you know.
I understand that third-party plugins might rely on those methods. The will have to deal with the fact that they will need to improve their behavior and start relying on corresponding IBuildSupport.